Lares/Assets/Mechanics/Inventory/UI/InventoryUIController.cs

```diff
@@ -0,0 +1,125 @@
+        [SerializeField] private Button _keyButton;
+
+        private InventoryNode _currentInventoryNode = null;
+
+        private void Start()
+        {
+            InventoryItemButton.ItemSelected += SetCurrentSelected;
+            _onUseButton.onClick.AddListener(() => { CallOnUseFunction(); });
+            _consumableButton.onClick.AddListener(() => { SetToConsumableView(); });
+            _materialsButton.onClick.AddListener(() => { SetToMaterialView(); });
+            _keyButton.onClick.AddListener(() => { SetToKeyView(); });
+            _consumableContentHolder = _consumableScrollView.GetComponent<ScrollRect>().content.transform;
+            _materialContentHolder = _materialScrollView.GetComponent<ScrollRect>().content.transform;
+            _keyContentHolder = _keyScrollView.GetComponent<ScrollRect>().content.transform;
+
+            LoadInventory();
+        }
+
+        private void SetToConsumableView()
+        {
+            _consumableScrollView.gameObject.SetActive(true);
+            _materialScrollView.gameObject.SetActive(false);
+            _keyScrollView.gameObject.SetActive(false);
+        }
+
+        private void SetToMaterialView()
+        {
+            _consumableScrollView.gameObject.SetActive(false);
+            _materialScrollView.gameObject.SetActive(true);
+            _keyScrollView.gameObject.SetActive(false);
+        }
+
+        private void SetToKeyView()
+        {
+            _consumableScrollView.gameObject.SetActive(false);
+            _materialScrollView.gameObject.SetActive(false);
+            _keyScrollView.gameObject.SetActive(true);
+        }
+
+        private void SetCurrentSelected(InventoryNode item)
+        {
+            _currentInventoryNode = item;
+            _itemName.text = item.Item.ItemData.ItemName;
+            _itemDescriptionText.text = item.Item.ItemData.ItemDescription;
+        }
+
+        private void CallOnUseFunction()
+        {
+            Debug.Log("On Use button clicked");
+            _currentInventoryNode.Item.OnUse(_inventory.PlayerReference);
+        }
+
+        void LoadInventory()
+        {
+            _inventoryList = new List<InventoryNode>();
+            _inventoryList = _inventory.InventoryData;
+            
+            GameObject temp;
+            foreach (InventoryNode item in _inventoryList)
+            {
+                temp = Instantiate(_inventoryItemButtonPrefab);
+                temp.GetComponent<InventoryItemButton>().ItemData = item;
+
+                if (item.Item is IConsumable)
+                {
+                    temp.transform.SetParent(_consumableContentHolder);
+                }
+                else if (item.Item is IMaterial)
+                {
+                    temp.transform.SetParent(_materialContentHolder);
+                }
+                else if (item.Item is IKey)
+                {
+                    temp.transform.SetParent(_keyContentHolder);
+                }
```

Comment:
`switch` cases can do the same as this.
```suggestion
                switch (item.Item)
                {
                    case IConsumable:
                        temp.transform.SetParent(_consumableContentHolder);
                        break;
                    case IMaterial:
                        temp.transform.SetParent(_materialContentHolder);
                        break;
                    case IKey:
                        temp.transform.SetParent(_keyContentHolder);
                        break;
                }
```